src/codegen.cpp

```diff
@@ -4055,32 +4055,17 @@ static Function *emit_function(jl_lambda_info_t *lam)
     // look for initial (line num filename) node
     if (jl_is_linenode(stmt)) {
         lno = jl_linenode_line(stmt);
-    }
-    else if (jl_is_expr(stmt) && ((jl_expr_t*)stmt)->head == line_sym &&
-             jl_array_dim0(((jl_expr_t*)stmt)->args) > 0) {
-        jl_value_t *a1 = jl_exprarg(stmt,0);
-        if (jl_is_long(a1))
-            lno = jl_unbox_long(a1);
-        if (jl_array_dim0(((jl_expr_t*)stmt)->args) > 1) {
-            a1 = jl_exprarg(stmt,1);
-            if (jl_is_symbol(a1))
-                filename = ((jl_sym_t*)a1)->name;
-            if (jl_array_dim0(((jl_expr_t*)stmt)->args) > 2) {
-                a1 = jl_exprarg(stmt,2);
-                if (jl_is_symbol(a1))
-                    dbgFuncName = ((jl_sym_t*)a1)->name;
-            }
-        }
+        filename = jl_linenode_file(stmt)->name;
     }
     ctx.lineno = lno;
 
     DIBuilder dbuilder(*m);
     ctx.dbuilder = &dbuilder;
 #ifdef LLVM37
-    DIFile *fil = NULL;
+    DIFile *topfil = NULL;
```

Comment:
style nit: i would prefer `topfile` here so you don't have to wonder if and how you abbreviated the name (and the reader doesn't have to wonder what FIL means here). plus vowels are free.